src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -580,6 +587,15 @@ def _to_data_or_iter(
             data_or_iter = (
                 iter(results_cursor) if to_iter else results_cursor.fetchall()
             )
+        # return limit number of rows
+        if show_limit is not None:
+            data_or_iter_result = []
+            for i, d in enumerate(data_or_iter):
+                if i >= show_limit:
+                    break
+                else:
+                    data_or_iter_result.append(d)
```

Comment:
I think we should update `result_set_to_rows` instead of making changes here. Can you take a look into that function and see if you also think that is a more appropriate place to make these changes.

I think we should get results using `to_iter=True` and parse results in `result_set_to_rows`